### DISCUSSION TITLE
Migrate monorepo to Yarn Modern or NPM workspaces

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [gatsbyjs/gatsby#33899] [Ideas / Feature Requests] [answered]
Is there an existing issue or discussion about upgrading this repository from Yarn 1 to Yarn Modern (Yarn 2, Yarn 3) or NPM workspaces (NPM 7, NPM 8)? I've searched a bunch, yet I can't seem to find any open issues/discussions on the topic.

Yarn v1 is not really suitable for use at this scale anymore, and has two viable replacements.

I believe it's the only reason the tests are so flaky; it's really a shame to have so many quality tests in this repo without actually being able to learn much from their failure. It's a huge blocker to community contributions.

Yarn 3 in particular is robust enough that I don't think a migration would be too painful anymore, whereas I expect it would significantly improve both the speed and reliability of the core dev experience.

I'd be happy to help out and/or make recommendations, however I do want to know first if there's any existing initiative, any major opposition to upgrading from the current broken package manager, or any strong opinion on returning to NPM vs. moving to Yarn Modern.

**Comment by LekoArts** [comment, marked as answer]
You can follow or help the PR from @wardpeet here: https://github.com/gatsbyjs/gatsby/pull/32937

**Reply by aaronadamsCA** [reply]
Terrific, thank you @LekoArts. @wardpeet, please let me know if there's anything I can help with.

Side note, the Gatsby Cloud team could probably use an internal nudge RE: supporting Yarn 3 as well. [I know it's on their radar but the feature request isn't even on the roadmap yet.](https://gatsby.canny.io/gatsby-cloud/p/support-yarn-2-and-pnp-api)
